### DISCUSSION TITLE
An attempt to enable the plugin fails with a "plugin not found"

### DISCUSSION THREAD

**Discussion by KondratevAD** [rabbitmq/rabbitmq-delayed-message-exchange#187] [General]
Help, I can't enable on the plugin, returns an error:
```
bash-5.1# rabbitmq-plugins enable rabbitmq_delayed_message_exchange-3.9.0.ez 
Enabling plugins on node rabbit@rabbit:
rabbitmq_delayed_message_exchange-3.9.0.ez
Error:
{:plugins_not_found, [:"rabbitmq_delayed_message_exchange-3.9.0.ez"]}
```
Uploaded the plugin to the directory `/opt/rabbitmq/plugins`:
```
bash-5.1# rabbitmq-plugins directories
Listing plugin directories used by node rabbit@rabbit
Plugin archives directory: /opt/rabbitmq/plugins
Plugin expansion directory: /var/lib/rabbitmq/mnesia/rabbit@rabbit-plugins-expand
Enabled plugins file: /etc/rabbitmq/enabled_plugins
```
Rabbitmq-plugins list:
```
Listing plugins with pattern ".*" ...
 Configured: E = explicitly enabled; e = implicitly enabled
 | Status: * = running on rabbit@rabbit
 |/
[  ] rabbitmq_amqp1_0                  3.9.13
[  ] rabbitmq_auth_backend_cache       3.9.13
[  ] rabbitmq_auth_backend_http        3.9.13
[  ] rabbitmq_auth_backend_ldap        3.9.13
[  ] rabbitmq_auth_backend_oauth2      3.9.13
[  ] rabbitmq_auth_mechanism_ssl       3.9.13
[  ] rabbitmq_consistent_hash_exchange 3.9.13
[  ] rabbitmq_delayed_message_exchange 3.9.0
[  ] rabbitmq_event_exchange           3.9.13
[  ] rabbitmq_federation               3.9.13
[  ] rabbitmq_federation_management    3.9.13
[  ] rabbitmq_jms_topic_exchange       3.9.13
[E*] rabbitmq_management               3.9.13
```
Rabbit version:
```
RabbitMQ version: 3.9.13
Erlang: 24.2
```

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
We cannot suggest much without knowing what exactly you did to "install" the .ez archive. One very common reason is the lack of file permissions on the downloaded file. If a RabbitMQ node cannot read a plugin or dependency file, the plugin won't be functional at some point (although it may be discoverable, depending on the permissions and operations performed by the node).

Running this node under `strace` will reveal a lot of relevant details.

**Comment by michaelklishin** [comment]
…and so will inspecting the permissions of `/opt/rabbitmq/plugins` (which is a non-standard directory, so you must be provisioning RabbitMQ from the generic binary tarball and thus are responsible for setting everything up).

**Comment by michaelklishin** [comment]
A quick GitHub search suggests someone failed to unzip the file served by GitHub: https://github.com/rabbitmq/rabbitmq-delayed-message-exchange/issues/104. That won't work either, the plugin archive directory must include `.ez` files (which are zip archives themselves that follow certain conventions). Then the node will [expand them on boot](https://rabbitmq.com/plugins.html#plugin-expansion) but it won't do that with a `.zip` file served by GitHub releases.

**Comment by michaelklishin** [comment]
Apparently we already have a couple of [doc sections](https://rabbitmq.com/plugins.html#troubleshooting-plugin-not-found) in the Plugins guide that cover all of the above.
